### PR TITLE
Use immer for reducers and reuse exisiting objects.

### PR DESCRIPTION
**Pull request by yuchenshi**
Removes the crazy rerendering for Firestore + RTDB.

### CODE REVIEW

File: src/store/config/reducer.ts [hunk 1]
```diff
@@ -14,18 +14,30 @@
  * limitations under the License.
  */
 
+import produce, { isDraft, original } from 'immer';
```
Comment (tjlav5) [inline src/store/config/reducer.ts]:
```suggestion
import produce from 'immer';
```

File: src/store/database/reducer.ts [hunk 2]
```diff
@@ -14,8 +14,11 @@
  * limitations under the License.
  */
 
+import produce from 'immer';
+import deepEquals from 'lodash.isequal';
```
Comment (tjlav5) [inline src/store/database/reducer.ts]:
```suggestion
```

**Comment by yuchenshi** [comment]
@tjlav5 I'm aware of this and I'll look into this. Merging this PR though to get GUI back to a usable (testable) state for now.

**Comment by yuchenshi** [comment]
@tjlav5 Fix coming in #121

**Pull request merged**